earth_enterprise/src/common/khCache.h

```diff
@@ -172,14 +186,51 @@ class khCache {
 #ifdef SUPPORT_VERBOSE
               verbose(verbose_),
 #endif
-              numItems(0)
+              numItems(0), cacheMemoryUse(0),
+              limitCacheMemory(false)
 
   {
     CheckListInvariant();
   }
   ~khCache(void) {
     clear();
   }
+  template<class T>
+  uint64 getKeyOrValueSize(const T obj) {
+    return sizeof(obj);
+  }
+  uint64 getKeyOrValueSize(const std::string str) {
+    return (sizeof(str) + (str.capacity() * sizeof(char)));
+  }
+  uint64 getKeyOrValueSize(const SharedString shstr) {
+    return shstr.GetSharedStringSize();
```

Comment:
Minor: shared strings don't include a pointer, so for this you can just `return sizeof(shstr)`. It won't make a huge difference (maybe no difference at all), but it might be a little more accurate.